Update beginner's guide to docs

stumblefiend opened the pull request:
beginners-guide-to-docs.rst is very specific to the developer audience, specifically for developers that are releasing their own open-source code, but the doc wasn't clearly labeled/branded that way. It is easy for visitors to confuse the purpose of the content with the other two "getting started" or "starting a doc" pages which I will also be updating soon. Also, it read like a presentation instead of feeling clear, actionable, and to the point. This commit updates the content so it is 100% obvious, at a glance, what the content is about and who the audience is.

Some areas of the content were edited to sound less like a manifesto. Also, the commit tries to suppress the presentation-style rhetoric to streamline for a written conversation that gets to the point faster.

Sentences were shortened for clarity and content was reorganized.

Also, keywords analysis was factored in for the page. Upon auditing for SEO, the keyword "code documentation" came up as an area the content should focus on capturing based on the volume of searches and the perceived difficulty of ranking for that keyword. Here is a breakdown of keywords analyzed for this specific page before creating this commit:

keyword | rank | monthly search volume | difficulty
-- | -- | -- | --
code documentation | 40 | 590 | 37
document write | 33 | 590 | 58
what documentation | 85 | 480 | 51
what is documentation | 75 | 480 | 39
create documentation | 42 | 390 | 58
doc writer | 34 | 390 | 55
document writer | 38 | 390 | 37
documentation example | 95 | 390 | 51
write document | 15 | 320 | 85
writing document | 7 | 320 | 78
how to document code | 9 |   |  
how to document | 61 | 260 | 52
write a document | 50 | 260 | 82
writing guide | 83 | 260 | 39
writing documentation | 1 | 210 | 45
documentation as code | 71 | 170 | 37
writing a document | 9 | 170 | 56
document writing | 3 | 140 | 65
documentation best practices | 79 | 140 | 31
documentaton | 55 | 140 | 81
how to doc | 56 | 140 | 40
how to write technical documentation | 69 | 140 | 47
prepare documentation | 3 | 140 | 43
proper documentation | 49 | 140 | 48
readme.md template | 77 | 140 | 46
written documentation | 2 | 140 | 31
code documentation example | 2 | 110 | 32
code documentation example | 59 | 110 | 32
code documentation template | 47 | 110 | 22
documentation skills | 90 | 110 | 22
how to write documentation | 2 | 110 | 55
document code | 33 | 90 | 37
documenting code | 24 | 90 | 42
how to document code | 15 | 90 | 40
write documentation | 1 | 90 | 53
best documentation | 53 | 70 | 47
document guide | 14 | 70 | 60
documentation creation | 44 | 70 | 41
formal documentation | 94 | 70 | 22
how to write a document | 3 | 70 | 57
how to write a how to guide | 45 | 70 | 40
how to write documentation for code | 2 | 70 | 40
html documentation example | 93 | 70 | 71
what is code documentation | 55 | 70 | 28
code documentation best practices | 30 | 50 | 30
docs writing | 14 | 50 | 76
documentation how to | 3 | 50 | 52
documentation in code | 35 | 50 | 45
documentation page | 74 | 50 | 56
documents writing | 2 | 50 | 65
how to documents | 71 | 50 | 46
how to write a documentation | 2 | 50 | 41
write manuals | 88 | 50 | 37
create software documentation | 81 | 40 | 38
how to write code documentation | 2 | 40 | 39
how to.write | 90 | 40 | 67
simple documentation | 2 | 40 | 25
technical documentation writing | 82 | 40 | 57
code documentation software | 66 | 30 | 26
code documentation tool | 94 | 30 | 32
how to create a documentation for project | 29 | 30 | 35
in code documentation | 32 | 30 | 33
write the document | 17 | 30 | 75
writing documentation for software | 15 | 30 | 35
writing software documentation | 5 | 30 | 43



<!-- readthedocs-preview writethedocs-www start -->
----
📚 Documentation preview 📚: https://writethedocs-www--2174.org.readthedocs.build/

<!-- readthedocs-preview writethedocs-www end -->

stumblefiend committed:
Update beginners-guide-to-docs.rst

beginners-guide-to-docs.rst is very specific to the developer audience but wasn't clearly labeled/branded that way. Also, it read like a presentation instead of feeling clear, actionable, and to the point. This commit updates the content so it is 100% obvious, at a glance, what the content is about and who the audience is.

Some areas of the content were edited to sound less like a manifesto. Also, the commit tries to suppress the presentation-style rhetoric to streamline for a written conversation that gets to the point faster.

Sentences were shortened for clarity and content was reorganized.

stumblefiend committed:
Update beginners-guide-to-docs.rst

CollierCZ commented:
When I see `code documentation`, I think of code comments, something this guide doesn't touch on at all. It seems like the page already ranks high (as high as 1) in the terms I'd expect, like `write documentation` and `writing software documentation`. I'd worry two things:
- By trying to reach a different keyword, we'd lose out on the high rank that is already there
- The new keyword isn't completely reflective of the content, which might hurt SEO in the long run.

I think the changes are good generally (except I'd be in large favor of removing the recommendation for an FAQ section, which I have many feelings about). I just don't know about `code documentation`. The current top results all seem to match my expectations of involving code comments/architecture decisions.

Perhaps we could take an inspiration from the listed next steps here and focus on documenting open-source projects? Or focus on documenting code projects or something like that?

ericholscher commented:
Yea, I think "open source documentation" or "software documentation" is probably a better framing than "code documentation" :+1:

stumblefiend commented:
> Yea, I think "open source documentation" or "software documentation" is probably a better framing than "code documentation" 👍

Thanks for stepping in to provide a phrase since this guide wasn't just a generic "how to write documentation" guide and was obviously catered to a more specific audience and purpose!

stumblefiend commented:
> When I see `code documentation`, I think of code comments, something this guide doesn't touch on at all. It seems like the page already ranks high (as high as 1) in the terms I'd expect, like `write documentation` and `writing software documentation`. I'd worry two things:
> 
> * By trying to reach a different keyword, we'd lose out on the high rank that is already there
> * The new keyword isn't completely reflective of the content, which might hurt SEO in the long run.
> 
> I think the changes are good generally (except I'd be in large favor of removing the recommendation for an FAQ section, which I have many feelings about). I just don't know about `code documentation`. The current top results all seem to match my expectations of involving code comments/architecture decisions.
> 
> Perhaps we could take an inspiration from the listed next steps here and focus on documenting open-source projects? Or focus on documenting code projects or something like that?

I agree 75% on the FAQ. My first reaction was also to remove that section since, in general, telling someone to create a FAQ is bad advice, but I then decided to keep this section because, for a starter documentation effort like in the case for a new project where the code maintainer is the sole documentarian, it actually makes initial sense. If we do keep the section, I'd feel more comfortable if we add a warning not to let FAQs become a band-aid doc where you throw in a bunch of content until you have some massive, unorganized list of questions and answers that are all just arbitrarily made up and aren't even real, "frequent asked questions" in the first place. But, if the dangers of FAQs always outweigh their limited utility, why even mention at all right? 

However, if we don't say anything about FAQs, we can't educate the audience on their dangers and, hence, will guarantee that people use them since they are such a tempting, easy fix. I vote for keeping the section but warning about their shortcomings so that people learn just how much of crappy solution they are. 

What are other's final thoughts on the FAQ section?

stumblefiend committed:
update headers from "code" to "software"

CollierCZ commented:
I think keeping it with a warning is a fine solution for now.

stumblefiend committed:
Merge branch 'main' into stumblefiend-patch-1

plaindocs commented:
@stumblefiend I'm not sure how In Progress this is, ping me if you need something specific to push it over the line

stumblefiend committed:
Merge branch 'main' into stumblefiend-patch-1

stumblefiend committed:
Update beginners-guide-to-docs.rst

Remove the FAQ section and mention the drawbacks of FAQs in a different, related section.

stumblefiend committed:
clarify the audience

Clarify the audience and area of focus in the first sentence of the doc.

stumblefiend commented:
@plaindocs I've made the final updates based on everyone's suggestions. This PR is ready for final review and approval.

stumblefiend committed:
Merge branch 'main' into stumblefiend-patch-1

plaindocs commented on docs/guide/writing/beginners-guide-to-docs.rst:
```diff
@@ -1,323 +1,210 @@
+---------------------------------------
 
-I am here to stop this scene from playing out.
+Understand your code in 6 months
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
This heading reads to me like "learn to code in six months" not "understand your own code six months from now"

plaindocs commented on docs/guide/writing/beginners-guide-to-docs.rst:
```diff
@@ -1,323 +1,210 @@
-At the end,
-you should have a project that is ready for public release.
+Your code from 6 months ago looks like code that someone else wrote.
+Things once obvious are now unclear and you now empathize with potential 
```
```suggestion
Things that were obvious six months ago are now unclear, and you now empathize with potential 
```

plaindocs commented on docs/guide/writing/beginners-guide-to-docs.rst:
```diff
@@ -1,323 +1,210 @@
 
-It's really easy to have an idea in your head that sounds perfect,
-but the act of putting words to paper requires a distillation of thought that may not be so easy.
+Few people will source dive to use code, but most 
```
rephrase "source dive" maybe?

plaindocs replied:
And I feel that document your project is different to document your code, but :man_shrugging:

plaindocs commented on docs/guide/writing/beginners-guide-to-docs.rst:
```diff
@@ -1,323 +1,210 @@
 
-.. _start your project with a README: http://tom.preston-werner.com/2010/08/23/readme-driven-development.html
+Requests_ is a great example.
 
-.. _write:
+.. _Requests: https://requests.kennethreitz.org/en/master/
 
-What to write
--------------
+Link to your code and issue tracker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Now we're getting down to the brass tacks.
-Making sure that you give your users all the information that they need,
-but not too much.
+People sometimes browse the code. They might want to file bugs for issues they've found, 
+so make your code easy to contribute. 
 
-First, you need to ask yourself who you're writing for.
-At first,
-you generally just need to appeal to two audiences:
+The `Python Guide`_ does a good job of this.
 
-* Users
-* Developers
+.. _Python Guide: http://docs.python-guide.org/en/latest/index.html
```
I like this framing. Do we want to go as far as pulling the example out and showing it here? Or are we happy just linking?

plaindocs commented on docs/guide/writing/beginners-guide-to-docs.rst:
```diff
@@ -1,323 +1,210 @@
-and don't care how it works.
-Developers are people who want to contribute back to your code.
+Tell how to get support
+~~~~~~~~~~~~~~~~~~~~~~~~
```
```suggestion
Tell people how to get support
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

plaindocs reviewed (approved):
Some comments, but this is much better than what we have.

stumblefiend committed:
Update docs/guide/writing/beginners-guide-to-docs.rst

Co-authored-by: Sam <sam@writethedocs.org>

stumblefiend committed:
Update docs/guide/writing/beginners-guide-to-docs.rst

Co-authored-by: Sam <sam@writethedocs.org>

stumblefiend committed:
Update beginners-guide-to-docs.rst

stumblefiend committed:
Merge branch 'main' into stumblefiend-patch-1
